### ISSUE TITLE
Error running with Java 23 - com.ibm.wala.ipa.cha.ClassHierarchyException: failed to load root <Primordial,Ljava/lang/Object> of class hierarchy

### ISSUE THREAD

**Issue by scottkurz** [IBM/codenet-minerva-code-analyzer#70]
Using v1.0.2 and a simple sample like: https://github.com/IBM/sample-app-mod and this

`java -jar codeanalyzer.jar -i C:\apps\sample-app-mod -o .... -a 2 --no-build`

Using a Java 23 JDK (Temurin 23.0.1)  both to point to 'java' AND set as JAVA_HOME, I get:

```
java.lang.RuntimeException: com.ibm.wala.ipa.cha.ClassHierarchyException: failed to load root <Primordial,Ljava/lang/Object> of class hierarchy
	at com.ibm.cldk.CodeAnalyzer.run(CodeAnalyzer.java:106)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1767)
	at picocli.CommandLine.access$900(CommandLine.java:145)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2139)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2106)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1973)
	at picocli.CommandLine.execute(CommandLine.java:1902)
	at com.ibm.cldk.CodeAnalyzer.main(CodeAnalyzer.java:95)
Caused by: com.ibm.wala.ipa.cha.ClassHierarchyException: failed to load root <Primordial,Ljava/lang/Object> of class hierarchy
	at com.ibm.wala.ipa.cha.ClassHierarchy.<init>(ClassHierarchy.java:300)
	at com.ibm.wala.ipa.cha.ClassHierarchy.<init>(ClassHierarchy.java:204)
	at com.ibm.wala.ipa.cha.ClassHierarchyFactory.make(ClassHierarchyFactory.java:87)
	at com.ibm.wala.ipa.cha.ClassHierarchyFactory.make(ClassHierarchyFactory.java:73)
	at com.ibm.cldk.SystemDependencyGraph.construct(SystemDependencyGraph.java:209)
	at com.ibm.cldk.CodeAnalyzer.analyze(CodeAnalyzer.java:185)
	at com.ibm.cldk.CodeAnalyzer.run(CodeAnalyzer.java:104)
	... 7 more
```

Using a Temurin v21 to do the same it succeeds.

**Comment by rahlk** [comment]
As of now WALA doesn’t support Java 23.

**Comment by scottkurz** [comment]
> ify if this issue persists in release 1.0.4

Yes this seems fixed in v1.0.4 (using Temurin v23.0.1).  Thank you !

**Comment by rahlk** [comment]
Thanks scott, closing this issue for now.

cc @cherylking, seems like java 23 is working
